apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KappControllerConfigSpec defines the desired state of KappControllerConfig
+type KappControllerConfigSpec struct {
+	// The namespace in which calico is deployed
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=kube-system
+	Namespace string `json:"namespace,omitempty"`
+
+	KappController KappController `json:"kappController,omitempty"`
+}
+
+type KappController struct {
+	// Whether to create namespace specified for kapp-controller
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	CreateNamespace bool `json:"createNamespace,omitempty"`
+
+	// The namespace value used for global packaging resources. Any Package and PackageMetadata CRs within that namespace will be included in all other namespaces on the cluster, without duplicating them
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	GlobalNamespace string `json:"globalNamespace,omitempty"`
+
+	Deployment KappDeployment `json:"deployment,omitempty"`
+
+	Config KappConfig `json:"config,omitempty"`
+}
+
+type KappDeployment struct {
+	// Whether to enable host networking for kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+
+	// The priority value that various system components use to find the priority of the kapp-controller pod
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// Concurrency of kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=4
+	Concurrency int `json:"concurrency,omitempty"`
+
+	// kapp-controller deployment tolerations
+	// +kubebuilder:validation:Optional
+	Tolerations []map[string]string `json:"tolerations,omitempty"`
+
+	// Bind port for kapp-controller API
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=10350
+	ApiPort int `json:"apiPort,omitempty"`
+
+	// Address for metrics server
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=:8080
+	MetricsBindAddress string `json:"metricsBindAddress,omitempty"`
+}
+
+type KappConfig struct {
+	// A cert chain of trusted CA certs. These will be added to the system-wide cert pool of trusted CA's
+	// +kubebuilder:validation:Optional
+	CaCerts string `json:"caCerts,omitempty"`
+
+	// The url/ip of a proxy for kapp controller to use when making network requests
+	// +kubebuilder:validation:Optional
+	HttpProxy string `json:"httpProxy,omitempty"`
```

Comment:
(non-blocking): proxy config for all packages needs to be consolidated https://github.com/vmware-tanzu/tkg-devel/issues/238

Reply:
TODO added